container/go/pkg/compat/image.go

```diff
@@ -0,0 +1,155 @@
+/// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Image for intermediate format used in python containerregistry.
+// Adopted from go-containerregistry's layout.image implementation with modification to understand rules_docker's legacy intermediate format.
+// Uses the go-containerregistry API as backend.
+
+package compat
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+// legacyImage is the image in legacy intermediate format. Implements v1.Image, and its implementation is very similar to layout.layoutImage.
+type legacyImage struct {
+	// path is the path to the directory containing the legacy image.
+	path string
+	// digest is the sha256 hash for this image.
+	digest v1.Hash
+	// manifestLock protects rawManifest.
+	manifestLock sync.Mutex
+	// rawManifest is the raw bytes of manifest.json file.
+	rawManifest []byte
+}
+
+var _ partial.CompressedImageCore = (*legacyImage)(nil)
+
+// MediaType of this image's manifest from manifest.json.
+func (li *legacyImage) MediaType() (types.MediaType, error) {
+	manifest, err := li.Manifest()
+	if err != nil {
+		return "", err
+	}
+
+	if !isExpectedMediaType(manifest.MediaType, types.OCIManifestSchema1, types.DockerManifestSchema2) {
```

Comment:
Given that there's only two media types and this check is only done once, I would remove this function as it hinders readability. Simply doing
`if manifest.MediaType != types.OCIManifestSchema1 && manifest.MediaType != types.DockerManifestSchema2`

is much more readable.